content/faq/errors.md

```diff
@@ -0,0 +1,38 @@
+### Common Errors
+
+During your development with NestJS, you may encounter a couple of errors as you learn the framework. 
+
+#### Cannot Resolve Dependency
+
+Probably the most common error message is about Nest not being able to resolve dependencies of a provider. The error message usually looks something like this:
+
+```bash
+Nest can't resolve dependencies of the <provider> (?). Please make sure that the argument <unknown_token> at index [<index>] is available in the <module> context.
+
+Potential solutions:
+- If <unknown_token> is a provider, is it part of the current <module>?
+- If <unknown_token> is exported from a separate @Module, is that module imported within <module>?
+  @Module({
+    imports: [ /* the Module containing <unknown_token> */ ]
+  })
+```
+
+There are a few gotchas, that are common. One is putting a provider in an `imports` array. If this is the case, the error will have the provider's name where `<module>` should be.
+
+#### Circular Dependency
+
+Occasionally you'll need to have [circular dependencies](/fundamentals/circular-dependency) in your application. Errors that arise from circular dependencies look like this:
```

Comment:
```suggestion
Occasionally you'll find it difficult to avoid [circular dependencies](/fundamentals/circular-dependency) in your application. You'll need to take some steps to help Nest resolve these. Errors that arise from circular dependencies look like this:
```